File: python/rlmesh/src/rlmesh/adapters/constants/parts.py
"""Suggested spellings for ``part``: the place on the body a role repeats at.

A ``part`` is an identity key on a leaf (``StateTag``, ``ImageTag``, ``Field``,
``Actuator``, ``Image``, ``State``): any identifier both sides agree on, matched
verbatim. Two arms are ``EEF_POS`` under ``part=LEFT_ARM`` and under
``part=RIGHT_ARM``. The constants below are conventions, not a registry: a part
outside them resolves the same way and draws no advisory.

Values are defined once, in the ``rlmesh-adapters`` crate (``roles/parts.rs``);
this module re-exports them through the native bindings.
"""

from ..._rlmesh import (
    BASE,
    HEAD,
    LEFT_ARM,
    LEFT_LEG,
    PARTS,
    RIGHT_ARM,
    RIGHT_LEG,
    TORSO,
)

__all__ = [
    "BASE",
    "HEAD",
    "LEFT_ARM",
    "LEFT_LEG",
    "PARTS",
    "RIGHT_ARM",
    "RIGHT_LEG",
    "TORSO",
]

File: crates/rlmesh-adapters/src/roles/parts.rs
//! Suggested part names: where a role repeats on a body.
//!
//! A `part` is any identifier both sides of a join agree on exactly; there is
//! no registry and nothing here validates one. These constants are
//! conventions for the common places a role repeats, nothing more.

pub const LEFT_ARM: &str = "left_arm";
pub const RIGHT_ARM: &str = "right_arm";
pub const HEAD: &str = "head";
pub const TORSO: &str = "torso";
pub const BASE: &str = "base";
pub const LEFT_LEG: &str = "left_leg";
pub const RIGHT_LEG: &str = "right_leg";

/// The suggested part names, for consumers exporting the conventions.
pub const PARTS: [&str; 7] = [LEFT_ARM, RIGHT_ARM, HEAD, TORSO, BASE, LEFT_LEG, RIGHT_LEG];
